### ISSUE TITLE
p-ColumnFilter : Change of translation for custom filter take in account too late (randomly).

### ISSUE THREAD

**Issue by Gwenou** [primefaces/primeng#14273]
### Describe the bug

When we use custom filter in the p-ColumnFilter we should pass the translation in the input matchModeOptions.

When the translation change we should change this value. For that we can subscribe on event like that in the parent component:
```
        this.translationSubscription = this.config.translationObserver.subscribe(() => {
            this.matchModeOptions =
                  customOption?.map((key: string) => {
                      return { label: this.translateService.instant("primeng." + key), value: key };
                  });
        });
```
But on the other side in the p-ColumnFilter the matchModeOptions is use in another subscribe:
```ts
        this.translationSubscription = this.config.translationObserver.subscribe(() => {
            this.generateMatchModeOptions();
            this.generateOperatorOptions();
        });

....

    generateMatchModeOptions() {
        this.matchModes =
            this.matchModeOptions ||
            (this.config as any).filterMatchModeOptions[this.type]?.map((key: any) => {
                return { label: this.config.getTranslation(key), value: key };
            });
    }
```

So if the subcribe in the p-ColumnFilter react in first the componement take the old value of the matchModeOptions  and store it in matchModes. The parent component subscribe react and take in account new langage)
=> Result : the change of the matchModeOptions is not take in account and translation appear as the previous langage selected.

In conclusion in p-ColumnFilter an ngOnChanges function is missing to react on the change of matchModeOptions:
```
  ngOnChanges(changes: SimpleChanges) {
    if (changes.matchModeOptions) {
      this.generateMatchModeOptions();
    }
  }
```

### Environment

I produce the bug localy on my dev environment.

### Angular version

16.2.10

### PrimeNG version

16.5.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

16.16.0

### Browser(s)

Edge

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you reopen the issue so we can include it in our roadmap?
Please don't forget to add your feedback as a comment after reopening the issue. These will be taken into account by us and will contribute to the development of this feature. Thanks a lot for your understanding!

Best Regards,
